### DISCUSSION TITLE
How to store user profile field in database when order is submitted?

### DISCUSSION THREAD

**Discussion by AkashMullick** [aannirajpatel/samosabucket#5] [Q&A] [answered]
We recently added a profile field for Venmo ID and now we are trying to modify the order process to save the user's Venmo ID in MongoDB as part of the order instead of the Stripe ID. We haven't been able properly load the venmo field from the user's profile. 

We made the following adjustments:
1) models/Order.js
Added this code to the OrderSchema:
`venmo: {
type: String,
  },` 

2) routes/order.js
Modified a line which passes tokens from the frontend by inserting venmo:
`const { cart, amount, address, venmo } = order;`
<br>Modified the order creation by adding `venmo: venmo,:`
`const order = new Order({
        userId: mongoose.Types.ObjectId(req.user.id),
        venmo: venmo,
        status: "PAID",
        cart: cart,
        amount: amount,
        delivery_address: address,
        delivery_time: deliveryTime,
        est_delivery_time: moment(deliveryTime).clone().add(30, "minutes"),
      });`

How do we correctly get the venmo field from the user's profile?

**Comment by aannirajpatel** [comment, marked as answer]
To access any data from a user's profile when you have the user's ID, you can call `User.findById`
If not already there, you might need to import the user schema, `User`, to the file in order to use it.

**Reply by AkashMullick** [reply]
Thanks for the suggestion! How would I retrieve a field called "venmo" from the user profile using the findById function? I tried adding `venmo: User.findById(mongoose.Types.ObjectId(req.user.id), 'venmo'),` to order in routes/order.js but that did not work.

**Reply by aannirajpatel** [reply]
You can use `await` for a User object when you do `User.findById(req.user.id)`, then access through the dot operator, as long as `venmo` is part of the schema.

**Reply by AkashMullick** [reply]
By any chance do you know the exact syntax to do this? I'm still learning React/MERN so I'm having some trouble.

**Reply by AkashMullick** [reply]
Nevermind, I got it to work using `const user = await User.findById(mongoose.Types.ObjectId(req.user.id));` and then `venmo: user.venmo,`. Thanks for the help!

**Reply by aannirajpatel** [reply]
Just saw this. Yes, that's the right way to do it.
